Caret API (#605)
codex-team/codex.editor: docs/api.md

`emit(eventName: string, data: object)` - fires all subscribed callbacks with passed data

### ListenerAPI

Methods that allows to work with DOM listener. Useful when you forgot to remove listener. Module collects all listeners and destroys automatically

`on(element: HTMLElement, eventType: string, handler: Function, useCapture?: boolean)` - add event listener to HTML element

`off(element: HTMLElement, eventType: string, handler: Function)` - remove event handler from HTML element

### CaretAPI

Methods to manage caret position.

Each method accept `position` and `offset` parameters. `Offset` should be used to shift caret by passed amount of characters.

> `Position` can be one of following values:

Reviewer: ```suggestion
`Position` can be one of the following values:
```

| Value     | Description 
| --------- | ----------- 

> | `start`   | Caret will be set at the Block start

Reviewer: ```suggestion
| `start`   | Caret will be set at the Block's beginning
```

| `end`     | Caret will be set at the Block end
| `default` | More or less emulates browser behaviour, in most cases behaves as `start`  

> Each method returns boolean value: true if caret is set successfully and false otherwise (e.g. when there is no Block at index);

Reviewer: ```suggestion
Each method returns `boolean` value: true if caret is set successfully or false otherwise (e.g. when there is no Block at index);
```
